sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -328,7 +328,8 @@ object StructType extends AbstractDataType {
   def apply(fields: Seq[StructField]): StructType = StructType(fields.toArray)
 
   def apply(fields: java.util.List[StructField]): StructType = {
-    StructType(fields.toArray.asInstanceOf[Array[StructField]])
+    val temp = new Array[StructField](0)
+    StructType(fields.toArray(temp))
```

Comment:
I think we can `import scala.collection.JavaConverters._` and use `StructType(fields.asScala)` here.